spicy/toolchain/src/compiler/codegen/unit-builder.cc

```diff
@@ -221,6 +226,21 @@ Type CodeGen::compileUnit(const type::Unit& unit, bool declare_only) {
     auto ft = _pb.parseMethodFunctionType({}, unit.meta());
     v.addField(type::struct_::Field(type::struct_::Field("__parse_stage1", std::move(ft))));
 
+    if ( auto convert = AttributeSet::find(unit.attributes(), "&convert") ) {
+        auto expression = *convert->valueAs<Expression>();
```

Comment:
I checked whether this works with the following example modified from the new docs,

```.ruby
    module Test;

    function foo(d: Data): int64 {
        return d.data.to_int();
    }

    type Data = unit {
        data: bytes &size=2;
    } &convert=foo(self);

    public type Foo = unit {
        numbers: Data[];

        on %done { print self.numbers; }
    };
```

This fails with the following error after codegen,

```
executing printf 12345678 | spicy-driver programming/examples/_parse-convert-unit.spicy

WARNING: execution failed: /var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/hilti-rt-test-0iOmjx/Test.cc:36:20: error: use of undeclared identifier 'foo'; did you mean 'Foo'?
            return foo(__self);
                   ^
/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/hilti-rt-test-0iOmjx/Test.cc:11:12: note: 'Foo' declared here
    struct Foo;
           ^
1 error generated.
[error] JIT compilation failed
[error] (no error output)
```

Reply:
Nice one, will fix.